### DISCUSSION TITLE
[How to] Anonymous users but prevent the anon key to easily access whole database

### DISCUSSION THREAD

**Discussion by nicolasbraun** [supabase/supabase#20356] [Questions] [answered]
Hi there,

I'm working on a service that do not require auth and the flow will be something like this (on a mobile app)
- User A creates something (let's say a shared note). `note` has a uuid in the db
- User A generates a share link (table `shares`)
- User B clicks the links, the app get's the associated note uuid, then the app get's the note content.

I do not want someone who get's the anon token to be able to call `/notes` and get all the notes.
Since my user are not authenticated I see 2 options that may be doable?

1. Use the anon token but somehow reject request on `notes` `shares` if a specific `WHERE uuid` clause is not in the request: I did not manage to do this through policies & functions
2. Maybe create user "silently" but this either require the admin token 😨  or maybe building an Edge function to do this (but it could be spammed)
3. Maybe I missed an easy way :)

Thanks for your help

N.

**Comment by GaryAustin1** [comment, marked as answer]
As far as your wanting to provide only a specific record and not the whole table.  I use an rpc call to a `security definer` type function.  That function takes a uuid in and then provides the row back.  Then on the table I enable RLS with no select policy.

**Reply by nicolasbraun** [reply]
Thanks @GaryAustin1 

Can you elaborate on how you can RPC call an SQL Function with the supabase API ?
Or are you creating an Edge Function that connects to the SQL database directly  and calls the function ?

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/guides/database/functions?language=js#simple-functions
And see call those postgres functions with js under the javascript tab with .rpc().

**Reply by nicolasbraun** [reply]
Thanks, I  was not finding the REST call for rpc in the [doc](https://supabase.com/docs/guides/database/functions) :)
Found it in my generated API Doc. Works like a charm.
